ProjectVoyage/View Controllers/Chat message/MessagesVC.m

```diff
@@ -0,0 +1,340 @@
+//
+//  MessagesVC.m
+//  Whatsapp
+//
+//  Created by Gui David
+//  Adapted from Rafael Castro
+//
+
+
+// Cells
+#import "MessageCell.h"
+
+// View controllers
+#import "MessagesVC.h"
+#import "EditMessageVC.h"
+#import "MarkdownExportVC.h"
+
+// Util
+#import "Util.h"
+
+// Helpers
+#import "Inputbar.h"
+#import "DAKeyboardControl.h"
+
+@interface MessagesVC() <InputbarDelegate,
+                                    UITableViewDataSource, UITableViewDelegate, ContainerProtocol, UITableViewDragDelegate, UITableViewDropDelegate>
+
+@property (nonatomic, strong) IBOutlet Inputbar *inputbar;
+@property (nonatomic, strong) IBOutlet UIBarButtonItem *markdownButton;
+
+@end
+
+#define UIColorFromRGB(rgbValue) [UIColor colorWithRed:((float)((rgbValue & 0xFF0000) >> 16))/255.0 green:((float)((rgbValue & 0xFF00) >> 8))/255.0 blue:((float)(rgbValue & 0xFF))/255.0 alpha:1.0]
+
+@implementation MessagesVC
+
+-(void)viewDidLoad
+{
+    [super viewDidLoad];
+    [self setInputbar];
+    [self setTableView];
+    //[self setGateway];
+    [self.tableView registerClass:MessageCell.class forCellReuseIdentifier:@"MessageCell"];
+}
+
+-(void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    
+    __weak Inputbar *inputbar = _inputbar;
+    __weak UITableView *tableView = _tableView;
+    __weak MessagesVC *controller = self;
+    
+    self.view.keyboardTriggerOffset = inputbar.frame.size.height;
+    [self.view addKeyboardPanningWithActionHandler:^(CGRect keyboardFrameInView, BOOL opening, BOOL closing) {
+        /*
+         Try not to call "self" inside this block (retain cycle).
+         But if you do, make sure to remove DAKeyboardControl
+         when you are done with the view controller by calling:
+         [self.view removeKeyboardControl];
+         */
+        
+        CGRect toolBarFrame = inputbar.frame;
+        toolBarFrame.origin.y = keyboardFrameInView.origin.y - toolBarFrame.size.height;
+        inputbar.frame = toolBarFrame;
+        
+        CGRect tableViewFrame = tableView.frame;
+        tableViewFrame.size.height = toolBarFrame.origin.y - 64;
+        tableView.frame = tableViewFrame;
```

Comment:
change these to strong objects within the completion block: its easier to do the weakSelf/strongSelf dance.

see the PFQuery example I posted earlier